FWCore/Framework/interface/CallbackExternalWork.h

```diff
@@ -0,0 +1,199 @@
+#include "FWCore/ServiceRegistry/interface/ServiceRegistry.h"
+#include "FWCore/ServiceRegistry/interface/ServiceToken.h"
+
+namespace edm {
+
+  class EventSetupImpl;
+
+  namespace eventsetup {
+
+    template <typename T,             //producer's type
+              typename TAcquireFunc,  //acquire functor type
+              typename TProduceFunc,  //produce functor type
+              typename TReturn,       //return type of the producer's method
+              typename TRecord,       //the record passed in as an argument
+              typename TDecorator     //allows customization using pre/post calls
+              = CallbackSimpleDecorator<TRecord>>
+    class CallbackExternalWork
+        : public CallbackBase<CallbackExternalWork<T, TAcquireFunc, TProduceFunc, TReturn, TRecord, TDecorator>,
+                              T,
+                              TProduceFunc,
+                              TReturn,
+                              TRecord,
+                              TDecorator> {
+    public:
+      using Base = CallbackBase<CallbackExternalWork, T, TProduceFunc, TReturn, TRecord, TDecorator>;
+
+      CallbackExternalWork(T* iProd, TProduceFunc iProduceFunc, unsigned int iID, const TDecorator& iDec = TDecorator())
+          : Base(iProd, std::make_shared<TProduceFunc>(std::move(iProduceFunc)), iID, iDec),
+            acquireFunction_(std::shared_ptr<TAcquireFunc>()) {}
+
+      CallbackExternalWork(T* iProd,
+                           TAcquireFunc iAcquireFunc,
+                           TProduceFunc iProduceFunc,
+                           unsigned int iID,
+                           const TDecorator& iDec = TDecorator())
+          : Base(iProd, std::make_shared<TProduceFunc>(std::move(iProduceFunc)), iID, iDec),
+            acquireFunction_(std::make_shared<TAcquireFunc>(std::move(iAcquireFunc))) {}
+
+      CallbackExternalWork(T* iProd,
+                           std::shared_ptr<TAcquireFunc> iAcquireFunc,
+                           std::shared_ptr<TProduceFunc> iProduceFunc,
+                           unsigned int iID,
+                           const TDecorator& iDec = TDecorator())
+          : Base(iProd, std::move(iProduceFunc), iID, iDec), acquireFunction_(std::move(iAcquireFunc)) {}
+
+      CallbackExternalWork* clone() {
+        return new CallbackExternalWork(
+            Base::producer(), acquireFunction_, Base::produceFunction(), Base::transitionID(), Base::decorator());
+      }
+
+      WaitingTaskHolder makeRunModuleTaskHolderImpl(oneapi::tbb::task_group* group,
+                                                    ServiceWeakToken const& token,
+                                                    EventSetupRecordImpl const* eventSetupRecordImpl,
+                                                    EventSetupImpl const* eventSetupImpl) {
+        bool emitPostPrefetchingSignal = true;
+        if (acquireFunction_) {
+          emitPostPrefetchingSignal = false;
+        }
```

Comment:
Now that `acquireFunction_` should always be set, could this become
```suggestion
        constexpr bool emitPostPrefetchingSignal = false;
```

Reply:
I agree. Done.